pkg/services/ngalert/provisioning/alert_rules.go

```diff
@@ -486,6 +662,21 @@ func (service *AlertRuleService) DeleteAlertRule(ctx context.Context, user ident
 	if storedProvenance != provenance && storedProvenance != models.ProvenanceNone {
 		return fmt.Errorf("cannot delete with provided provenance '%s', needs '%s'", provenance, storedProvenance)
 	}
+
+	can, err := service.authz.CanWriteAllRules(ctx, user)
+	if err != nil {
+		return err
+	}
+	if !can {
+		delta, err := store.CalculateRuleDelete(ctx, service.ruleStore, rule.GetKey())
+		if err != nil {
+			return err
+		}
+		if err = service.authz.AuthorizeRuleGroupWrite(ctx, user, delta); err != nil {
+			return err
+		}
+	}
+
 	return service.xact.InTransaction(ctx, func(ctx context.Context) error {
```

Comment:
```suggestion
	// The single delete is idempotent, and doesn't error when deleting a group that already doesn't exist.
	// This is different from deleting groups. We delete the rules directly rather than persisting a delta here to keep the semantics the same.
	// TODO: Either persist a delta here as a breaking change, or deprecate this endpoint in favor of the group endpoint.
	return service.xact.InTransaction(ctx, func(ctx context.Context) error {
```
There's a subtle choice here, I think this PR makes the right choice but now it's clear for readers in a comment. I had to so some significant archaeology to make sure this lined up correctly.